### ISSUE TITLE
Lost my iPhone and all the codes

### ISSUE THREAD

**Issue by rvilde** [mattrubin/authenticator#242]
I lost my iPhone and al the codes. How can I move them to my new iPhone? Any help would be appreciated! Thank you!

**Comment by beaucollins** [comment]
The only way to recover the tokens is if you have made an encrypted backup of your phone through iTunes. Backing up a phone in this way includes the encrypted data in the phone's keychain.

If you have an encrypted backup, restoring your new phone using this backup will give you the codes.

If you do _not_ have an encrypted backup, there is unfortunately no way to get the secrets. They are never transmitted by the Authenticator app in any way.

To restore access to the accounts protected by Authenticator you will have to go through the password/login recovery process that each of the websites provide. Usually you can get started by going to the login form for the website and clicking the "forgot password" link.

It's up to the website's staff to verify that you are the account owner and they all have different ways of doing this.

Best of luck!

**Comment by cosmoburn** [comment]
@beaucollins Could you clarify to me if Authenticator is using the `kSecAttrAccessibleThisDeviceOnly` mode to store entries in the system keychain, or if it is using a format that persists across devices?

**Comment by beaucollins** [comment]
`kSecAttrAccessibleThisDeviceOnly` is _not_ specified for the keychain entries that Authenticator creates.

@ArthurSilveira the keychain modification for this app happens through the [OneTimePassword library](https://github.com/mattrubin/OneTimePassword/blob/432f49691e880fe8baa755ad55e74133db02ad7e/Sources/Keychain.swift#L111-L163).


Keychain items for Authenticator will be included in an **encrypted** backup through iTunes. A phone that is restored using one of these backups will have the one-time password secrets restored as well because encrypted backups include all of the device's keychain entries. Since the keychain items do not include the `kSecAttrSynchronizable` attribute they will not be synced to an iCould keychain.

**Comment by cosmoburn** [comment]
Thank you for the reply @beaucollins 

So from what I gather, since I only backed up my phone through the automatic iCloud backup then I'm shit out of luck as far as getting these codes back haha.

**Comment by beaucollins** [comment]
@ArthurSilveira right. You'll have to recover access to your accounts through the respective websites that host your accounts if you no longer have access to the one time password generators.

**Comment by cosmoburn** [comment]
@beaucollins thank you very much, lesson learned heh.
